ve/opencl/engine_opencl.cpp

```diff
@@ -358,7 +366,10 @@ void EngineOpenCL::execute(const std::string &source, const std::vector<bh_base*
     auto texec = chrono::steady_clock::now();
     queue.enqueueNDRangeKernel(opencl_kernel, cl::NullRange, ranges.first, ranges.second);
     queue.finish();
-    stat.time_exec += chrono::steady_clock::now() - texec;
+    auto exec_duration = chrono::steady_clock::now() - texec;
+    stat.time_exec += exec_duration;
+    ++stat.time_per_kernel[source_filename].first;
+    stat.time_per_kernel[source_filename].second += exec_duration;
```

Comment:
Would the minimum and maximum execution time make sense to store as well?

Reply:
Would you expect it to fluctuate a lot? Should be straightforward to implement.

Reply:
I wouldn't expect that, but I believe it could be useful data.